server/alerts_email_via_smtp.py

```diff
@@ -0,0 +1,270 @@
+		self.render_template = king_phisher.templates.TemplateEnvironmentBase().from_string(self.email_template)
+		return True
+
+	def on_campaign_alert(self, table, alert_subscription, count):
+		user = alert_subscription.user
+		if not self.config['smtp_email']:
+			self.logger.debug("user {0} has no email address specified, skipping SMTP alert".format(user.id))
+			return False
+		if self.send_message(table, alert_subscription, count, self.config['smtp_email']):
+			return True
+		return False
+
+	def get_template_vars(self, table, alert_subscription, count):
+		campaign = alert_subscription.campaign
+		template_vars = {
+			'id': str(campaign.id),
+			'name': campaign.name,
+			'created': campaign.created,
+			'expiration': campaign.expiration,
+			'has_expired': campaign.has_expired,
+			'message_count': len(campaign.messages),
+			'visit_count': len(campaign.visits),
+			'credential_count': len(campaign.credentials),
+			'time': datetime.datetime.utcnow()
+		}
+		return template_vars
+
+	def create_headers(self, table, alert_subscription, count):
+
+		message = MIMEMultipart()
+		message['Subject'] = "Campaign Event: {0}".format(alert_subscription.campaign.name)
+		message['From'] = "<{0}>".format(self.config['smtp_email'])
+		message['To'] = "<{0}>".format(self.config['smtp_email'])
+
+		textual_message = MIMEMultipart('alternative')
+		txt_content = "{0:,} {1} reached for campaign: {2}".format(count, table.replace('_', ' '), alert_subscription.campaign.name)
+		plaintext_part = MIMEText(txt_content, 'plain')
+		textual_message.attach(plaintext_part)
+
+		try:
+			rendered_email = self.render_template.render(template_vars=self.get_template_vars(table, alert_subscription, count))
+		except:
+			self.logger.warning('failed to render email jinja template', exc_info=True)
+			return False
+		html_part = MIMEText(rendered_email, 'html')
+		textual_message.attach(html_part)
+
+		message.attach(textual_message)
+		encoded_email = message.as_string()
+		return encoded_email
+
+	def send_message(self, table, alert_subscription, count, target_email):
+		msg = self.create_headers(table, alert_subscription, count)
+		if not msg:
+			return False
+		source_email = self.config['smtp_email']
+		server = smtplib.SMTP(self.config['smtp_host'], self.config['smtp_port'])
+		if self.config.get('ssl', False):
+			try:
+				server = smtplib.SMTP_SSL(self.config['smtp_host'], self.config['smtp_port'])
+			except smtplib.SMTPException:
+				self.logger.warning('received an SMTPException while negotiating STARTTLS with the SMTP server', exc_info=True)
+				return False
+
+		try:
+			server.connect(self.config['smtp_host'], self.config['smtp_port'])
+			server.ehlo()
+		except smtplib.SMTPException:
+			self.logger.warning('received an SMTPException while connecting to SMTP server', exc_info=True)
+			return False
+
+		if not self.config.get('ssl', False) and 'starttls' in server.esmtp_features:
```

Comment:
If `starttls` is available, use it regardless of the `ssl` setting. `ssl` should be for explicit TLS at start, i.e. using `SMTP_SSL` for the connection.